ARCHITECTURE.md

```diff
@@ -158,10 +173,115 @@ This button with type submit inside a form will make it submit on ENTER key pres
+
+
+
+## Architecture Goal 
+
+The geyser-app is currently being restructured to improve maintainability, readability, and performance.
+Guided by Domain Driven Design (DDD) principles and the Horizontal Split principle, we'll reorganize all pages and components into Modules.
+
+### Project Structure
+The base structure of the app would be introduced with the following folders among the existing ones:
+- /src/shared
+- /src/modules
+
+#### Shared
+This would contain all of the business-logic-less ui components, contants, helpers and utils, that would be required by the modules.
+
+A general structure looks like:
+1. `shared` package
+    1. `components` 
+        1. `ui`
+        2. `layout`
+        3. `input`
+            1. `form`
+            2. `elements`
+    2. `constants`
+    3. `utils`
+
+#### Modules
+We will have a module per domain, and a domain would consist of multiple pages, along with other tools. Based on the nature of pages and the difference in complexity. We have `views` inside `pages` that replicate a `pages` structure, so we can go as deeply nested as needed.
+
+ A general structure for modules looks like:
+
+1.  `states` *( state manipulation / atoms )*
+2.  `hooks` *( logic, and communication backend )*
+3.   `components` *( ui and layout components )*
+4.  `graphQL` *( fragment, queries definition )*
+5.  `types` *( fragment types, query and mutation hooks )*
+6.  `pages` 
+	1.  `states`
+	2.  `hooks`
+	5.  `components`
+	6. ` views`
+        1.  *…recursive ( pages )*
+
+Below are the planned modules and the pages that would go in them based on the pages that are currently in the app. 
+
+1.  **project**
+	1. `projectView`
+	2. `projectCreate`
+	3. `projectDashboard`
+	4. `projectFunding`
+	5. `entry`
+2.  **discovery**
+	1. `landing`
+	2. `projectLeaderboard`
+3.  **grants**
+	1. `grantsLanding`
+	2. `grantsPage`
+4.  **profile**
+	1. `profilePage`
+	2. `profileSettings`
+5.  **auth**
+	1. `loginAuth`
+	2. `otp`
+6.  **general**
+	1. `badges`
+	2. `legal`
+	3. `about`
+	4. `fallback`
+
+### Guiding Principles
+
+These are some of the principles that drive us towards the architectural goal discussed above and is expected as we build geyser together:
+
+1. *React Components (.tsx) inside `pages/views`, would have entire focus on the UI, with invocation of hooks to support them.*
+
+	  This means, anything relating towards business-logic and state-management apart from low level boolean and store of value, will not be implemented inside the components, keeping UI components clean and easy to maintain.
+
+2. *All state management logic will house inside `states` folder at coresponding layer ( pages/views/sections), and we use Jotai for state management.*
```

Comment:
Here you can link to the Jotai section, which would either be below this section or in another md file, eg: `CODING_GUIDELINES.md`